numpy/core/src/_simd/_simd_inc_vector.h

```diff
@@ -0,0 +1,186 @@
+#if !NPY_SIMD
+    #error "Not a standalone header, only works through 'simd.dispatch.c.src'"
+#endif
+
+/************************************
+ ** Private Definitions
+ ************************************/
+// PySequenceMethods
+static Py_ssize_t
+simd__vector_length(PySIMDVectorObject *self)
+{
+    return simd_data_getinfo(self->dtype)->nlanes;
+}
+static PyObject *
+simd__vector_item(PySIMDVectorObject *self, Py_ssize_t i)
+{
+    const simd_data_info *info = simd_data_getinfo(self->dtype);
+    int nlanes = info->nlanes;
+    if (i >= nlanes) {
+        PyErr_SetString(PyExc_IndexError, "list index out of range");
+        return NULL;
+    }
+    npyv_lanetype_u8 *src = self->data + i * info->lane_size;
+    simd_data data;
+    memcpy(&data.u64, src, info->lane_size);
+    return simd_scalar_to_obj(data, info->to_scalar);
+}
+
+static PySequenceMethods simd__vector_as_sequence = {
+    (lenfunc) simd__vector_length,           /* sq_length */
+    (binaryfunc) NULL,                       /* sq_concat */
+    (ssizeargfunc) NULL,                     /* sq_repeat */
+    (ssizeargfunc) simd__vector_item,        /* sq_item */
+    (ssizessizeargfunc) NULL,                /* sq_slice */
+    (ssizeobjargproc) NULL,                  /* sq_ass_item */
+    (ssizessizeobjargproc) NULL,             /* sq_ass_slice */
+    (objobjproc) NULL,                       /* sq_contains */
+    (binaryfunc) NULL,                       /* sq_inplace_concat */
+    (ssizeargfunc) NULL,                     /* sq_inplace_repeat */
+};
+
+static PyObject *
+simd__vector_name(PySIMDVectorObject *self)
+{
+    return PyUnicode_FromString(simd_data_getinfo(self->dtype)->pyname);
+}
+static PyGetSetDef simd__vector_getset[] = {
+    { "__name__", (getter)simd__vector_name, NULL, NULL, NULL },
+    { NULL, NULL, NULL, NULL, NULL }
+};
+
+// PyTypeObject(PySIMDVectorType)
+static PyObject *
+simd__vector_repr(PyObject *self)
+{
+    // PySequence_Fast returns Tuple in PyPy
+    PyObject *obj = PySequence_List(self);
+    if (obj != NULL) {
+        PyObject *repr = PyObject_Str(obj);
```

Comment:
It would be good to include information about the type in the `repr`, perhaps:
```suggestion
        PyObject *repr = PyUnicode_FromFormat("<%s of %R>", NPY_TOSTRING(NPY_CPU_DISPATCH_CURFX(VECTOR)), obj);
```

Reply:
The nested classes for the testing unit already provides decent information,
however if I re-write it to:
```C
static PyObject *
simd__vector_repr(PySIMDVectorObject *self)
{
    // PySequence_Fast returns Tuple in PyPy
    PyObject *obj = PySequence_List((PyObject*)self);
    if (obj != NULL) {
        const char *type_name = simd_data_getinfo(self->dtype)->pyname;
        #ifdef NPY__CPU_TARGET_CURRENT
            const char *mod_name = NPY_TOSTRING(NPY_CPU_DISPATCH_CURFX(NPYV));
        #else
            const char *mod_name = "NPYV_BASELINE";
        #endif
        PyObject *repr = PyUnicode_FromFormat("<%s.%s of %R>", mod_name, type_name, obj);
        Py_DECREF(obj);
        return repr;
    }
    return obj;
}
```
will produce the following backtrace:
```Python
self = <numpy.core.tests.test_simd.Test_SIMD_ALL_128_SSE42_u64 object at 0x7f1935587470>

    def test_memory_load(self):
        data = self._data()
        # unaligned load
        load_data = self.load(data)
>       assert load_data != data
E       assert <NPYV_SSE42.npyv_u64 of [1, 2]> != [1, 2]

data       = [1, 2]
load_data  = <NPYV_SSE42.npyv_u64 of [1, 2]>
self       = <numpy.core.tests.test_simd.Test_SIMD_ALL_128_SSE42_u64 object at 0x7f1935587470>
```
I'm not sure what do you think? I feel it would be better if the assert line shorted
```Python
E       assert [1, 2] != [1, 2]
````

Reply:
I can also remove the prefix of the module and only provides the target name
```Python
E       assert <SSE42.npyv_u64 of [1, 2]> != [1, 2]
```

Reply:
As long as the repr makes it clear I'm not looking at a regular list, any of those are fine.

Reply:
I shorten it  to became :
```Python
E       assert (npyv_u64)[1, 2] != [1, 2]
````

Reply:
Typically one of two things is true about the repr:
* `eval(repr(x)) == x`
* `repr(x) == "<...>"`

So I'd prefer to see the angle brackets

Reply:
done